module/VuFind/src/VuFind/Db/Table/Comments.php

```diff
@@ -174,4 +174,55 @@ public function getStatistics()
         $result = $statement->execute();
         return (array)$result->current();
     }
+
+    /**
+     * Get a paginated result of all comments
+     * made by the user
+     *
+     * @param int    $userId User ID
+     * @param int    $limit  Limit
+     * @param int    $page   Page
+     * @param string $sort   Sort
+     *
+     * @return \Laminas\Paginator\Paginator
+     */
+    public function getCommentsPaginator(
+        int $userId,
+        int $limit,
+        int $page,
+        string $sort
+    ): \Laminas\Paginator\Paginator {
+        $commentSelect = new Select();
+        $commentSelect->from('comments')
+            ->where->equalTo('comments.user_id', $userId);
+        $commentSelect->columns([
+            'resource_id',
+            'id',
+            'comment',
+            'user_id',
+            'created',
+        ])
+        ->join(
+            ['re' => 'resource'],
+            'comments.resource_id = re.id',
+            ['record_id', 'source'],
+            Select::JOIN_LEFT
+        );
+        $order = $sort ? $sort : 'created DESC';
+        $commentSelect->order($order);
+        if ($page > 0) {
+            $commentSelect->offset($page);
+        }
+        if (null !== $limit) {
```

Comment:
Since `$limit` is typed to `int`, I don't think you need this null check; shouldn't be possible for the value to be null.